extensions/azure/cosmos/cosmos-common/src/main/java/org/eclipse/dataspaceconnector/azure/cosmos/dialect/WhereClause.java

```diff
@@ -70,32 +70,17 @@ String getWhere() {
         return where;
     }
 
-    private void criterion(Criterion criterion) {
-        if (!SUPPORTED_OPERATOR.contains(criterion.getOperator())) {
-            throw new IllegalArgumentException("Cannot build SqlParameter for operator: " + criterion.getOperator());
+    private void parse(Criterion criterion) {
+        var expr = new CosmosConditionExpression(criterion);
+        var exprResult = expr.isValidExpression();
+        if (exprResult.failed()) {
+            throw new IllegalArgumentException("Cannot build WHERE clause, reason: " + String.join(", ", exprResult.getFailureMessages()));
         }
-        var operandLeft = sanitize(criterion.getOperandLeft().toString());
-        var operandRight = criterion.getOperandRight().toString();
-        var queryParam = createQueryParam(operandLeft, criterion.getOperator(), operandRight);
-        String param = queryParam.isEmpty() ? operandRight : sanitizedParameterName(operandLeft);
-        where += parameters.isEmpty() ? "WHERE" : " AND";
-        parameters.addAll(queryParam);
-        where += String.format(" %s.%s %s %s", objectPrefix, operandLeft, criterion.getOperator(), param);
-    }
-
-    private List<SqlParameter> createQueryParam(String opLeft, String operator, String opRight) {
-        switch (operator) {
-            case EQUALS_OPERATOR:
-                return List.of(new SqlParameter(sanitizedParameterName(opLeft), opRight));
-            case IN_OPERATOR:
-                return List.of();
-            default: //not actually needed, but it's good style to have a default case
-                throw new IllegalArgumentException("Cannot build SqlParameter for operator: " + operator);
-        }
-    }
 
-    private String sanitizedParameterName(String operandLeft) {
-        return "@" + operandLeft.replaceAll("\\.", "");
+        var operandLeft = sanitize(criterion.getOperandLeft().toString());
+        where += where.startsWith("WHERE") ? " AND" : "WHERE"; //if we have a chained WHERE ... AND ... statement
+        parameters.addAll(expr.toParameters());
+        where += String.format(" %s.%s %s %s", objectPrefix, operandLeft, criterion.getOperator(), expr.toValuePlaceholder());
```

Comment:
The knowledge on how to format a condition should stay in the `CosmonConditionExpression` class, then the `WhereClause` could only get the list of conditions, transforming them to string and joining with the ` AND ` separator and `WHERE ` prefix.